modules/afsql/afsql.c

```diff
@@ -805,12 +807,85 @@ afsql_dd_build_insert_command(AFSqlDestDriver *self, LogMessage *msg, GString *t
             }
           else
             {
-              dbi_conn_quote_string_copy(self->dbi_ctx, value->str, &quoted);
+              TypeHint type = self->fields[i].value->type_hint;
```

Comment:
This function became hard to understand, can you factor out some parts of it to separate functions? Also as I can see the `else` (`need_drop` and `fallback`) parts are really similar. That logic might be factored out to one function, too.

Reply:
Sure @alltilla